base/tps/src/main/java/org/dogtagpki/server/tps/TPSSubsystem.java

```diff
@@ -101,8 +101,8 @@ public void init(ConfigStore config) throws Exception {
 
         logger.info("Initializing TPS subsystem");
 
-        org.dogtagpki.server.tps.TPSEngine engine = org.dogtagpki.server.tps.TPSEngine.getInstance();
-        TPSEngineConfig cs = engine.getConfig();
+        org.dogtagpki.server.tps.TPSEngine tpsEngine = (org.dogtagpki.server.tps.TPSEngine) engine;
```

Comment:
This now name shadows the field on line 83. At first I thought that because this class inherits it's engine instance from `Subsystem` the field is not necessary, but they are actually different instances, each of a different class sharing the name `TPSEngine`. This is a bit confusing!

Reply:
Yes, we can fix that separately later, but I'm not sure what we should rename it to. Do you have any suggestion?

Reply:
Does `org.dogtagpki.server.tps.engine.TPSEngine` need to exist at all? Can we merge it into  `org.dogtagpki.server.tps.TPSEngine`?

`org.dogtagpki.server.tps.engine.TPSEngine` has an empty constructor, it's `init` method does nothing and there is an `initialize` method that returns a `-1` error code. The class doesn't inherit or implement anything, so the other constants and methods could just move into `org.dogtagpki.server.tps.engine.TPSEngine`?

Reply:
https://github.com/dogtagpki/pki/pull/4362